addons/product/models/product_pricelist.py

```diff
@@ -619,30 +619,27 @@ def _is_applicable_for(self, product, qty_in_product_uom):
         if self.min_quantity and qty_in_product_uom < self.min_quantity:
             res = False
 
-        elif self.categ_id:
-            # Applied on a specific category
-            cat = product.categ_id
-            while cat:
-                if cat.id == self.categ_id.id:
-                    break
-                cat = cat.parent_id
-            if not cat:
+        elif self.applied_on == "2_product_category":
+            if (
+                product.categ_id != self.categ_id
```

Comment:
Note: I added this small little optimization to avoid reading `parent_path` if the categories are an exact match.
It has the potential to save a query in this corner case.